### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by Deng-Y** [ucbdrive/hd3#5]
1. I found two undefined variables `H, W` below. Can you fix it? What's the intuition behind the weight `4**(ds - l)`?
https://github.com/ucbdrive/hd3/blob/aefb9da9f2512299555c41255f199d3df7834546/hd3losses.py#L32

2. Why yield the parameters of the decoder twice in `HD3Model.optim_parameters()`, did you do it for a reason?
https://github.com/ucbdrive/hd3/blob/aefb9da9f2512299555c41255f199d3df7834546/hd3model.py#L51

3. Can you provide the training&evaluation loss curves of the stereo models for further analysis?

Thank you!

**Comment by yzcjtr** [comment]
Hi, I have fixed the missing variable issue in the latest commit. Also, I forgot to clean the optimizer code. Thanks for the reminder.

As for the weight, ``batchmean`` flag only averages the loss across the batch dimension but we want to average it over H and W dimensions as well. Note that different levels have different resolutions.

**Comment by Deng-Y** [comment]
Thanks for your answer. Can you provide the training&evaluation loss/EPE curves of the stereo models for further analysis?

**Comment by yzcjtr** [comment]
![Screen Shot 2019-06-10 at 3 26 15 PM](https://user-images.githubusercontent.com/2462939/59179313-390bb500-8b94-11e9-9c88-b7dc14c7c9e2.png)

Example EPE curve on the validation set of FlyingThings3D. FYI.

**Comment by Deng-Y** [comment]
Thanks again.
